ceremonies/src/lib.rs

```diff
@@ -56,293 +54,336 @@ use sp_std::{prelude::*, vec};
+			proof: Option<ProofOfAttendance<T::Signature, T::AccountId>>,
+		) -> DispatchResultWithPostInfo {
 			let sender = ensure_signed(origin)?;
-			ensure!(<encointer_scheduler::Module<T>>::current_phase() == CeremonyPhaseType::REGISTERING,
-				Error::<T>::RegisteringPhaseRequired);
+			ensure!(
+				<encointer_scheduler::Module<T>>::current_phase() == CeremonyPhaseType::REGISTERING,
+				Error::<T>::RegisteringPhaseRequired
+			);
 
-			ensure!(<encointer_communities::Pallet<T>>::community_identifiers().contains(&cid),
-				Error::<T>::InexistentCommunity);
+			ensure!(
+				<encointer_communities::Pallet<T>>::community_identifiers().contains(&cid),
+				Error::<T>::InexistentCommunity
+			);
 
 			let cindex = <encointer_scheduler::Module<T>>::current_ceremony_index();
 
 			if Self::is_registered(cid, cindex, &sender) {
-				return Err(<Error<T>>::ParticipantAlreadyRegistered.into());
+				return Err(<Error<T>>::ParticipantAlreadyRegistered.into())
 			}
 
 			if let Some(p) = &proof {
 				// we accept proofs from other communities as well. no need to ensure cid
 				ensure!(sender == p.prover_public, Error::<T>::WrongProofSubject);
 				ensure!(p.ceremony_index < cindex, Error::<T>::ProofAcausal);
-				ensure!(p.ceremony_index >= cindex.checked_sub(T::ReputationLifetime::get()).unwrap_or(0), Error::<T>::ProofOutdated);
-				ensure!(Self::participant_reputation(&(p.community_identifier, p.ceremony_index),
-					&p.attendee_public) == Reputation::VerifiedUnlinked,
-					Error::<T>::AttendanceUnverifiedOrAlreadyUsed);
+				ensure!(
+					p.ceremony_index >=
+						cindex.checked_sub(T::ReputationLifetime::get()).unwrap_or(0),
+					Error::<T>::ProofOutdated
+				);
+				ensure!(
+					Self::participant_reputation(
+						&(p.community_identifier, p.ceremony_index),
+						&p.attendee_public
+					) == Reputation::VerifiedUnlinked,
+					Error::<T>::AttendanceUnverifiedOrAlreadyUsed
+				);
 				if Self::verify_attendee_signature(p.clone()).is_err() {
-					return Err(<Error<T>>::BadProofOfAttendanceSignature.into());
+					return Err(<Error<T>>::BadProofOfAttendanceSignature.into())
 				};
 
 				// this reputation must now be burned so it can not be used again
-				<ParticipantReputation<T>>::insert(&(p.community_identifier, p.ceremony_index),
-					&p.attendee_public, Reputation::VerifiedLinked);
+				<ParticipantReputation<T>>::insert(
+					&(p.community_identifier, p.ceremony_index),
+					&p.attendee_public,
+					Reputation::VerifiedLinked,
+				);
 				// register participant as reputable
-				<ParticipantReputation<T>>::insert((cid, cindex),
-					&sender, Reputation::UnverifiedReputable);
+				<ParticipantReputation<T>>::insert(
+					(cid, cindex),
+					&sender,
+					Reputation::UnverifiedReputable,
+				);
 			};
 
 			Self::register(cid, cindex, &sender, proof.is_some())?;
 
 			debug!(target: LOG, "registered participant: {:?}", sender);
-			Ok(())
+
+			Self::deposit_event(Event::ParticipantRegistered(sender));
```

Comment:
With the new pallet syntax it was discovered that we never threw the event! This was probably accidentally deleted sometimes.

Reply:
Oh. I think we should use this event as it indicates real activity

Reply:
* I agree, the ceremonies-pallet is very quiet anyhow: #130